pkg/dumper/dump.go

```diff
@@ -198,10 +198,19 @@ func (c PostgresDumpHandler) Dump(ctx context.Context, r io.Reader, path string)
 		return err
 	}
 
+	//group them by run group identifier (line, direction, and trainid)
+	var runs = map[string][]martaapi.Schedule{}
 	for _, rec := range records {
-		err := c.upserter.AddRecordToDatabase(rec)
-		if err != nil {
-			c.logger.Error(fmt.Sprintf("failed to upsert MARTA API response to postgres: %s", err.Error()))
+		rgi := postgres.RunGroupIdentifierFor(martaapi.Direction(rec.Direction), martaapi.Line(rec.Line), rec.TrainID)
+		runs[rgi] = append(runs[rgi], rec)
+	}
+
+	for _, run := range runs {
+		for i := range run {
+			err := c.upserter.AddRecordToDatabase(run, i)
+			if err != nil {
+				c.logger.Error(fmt.Sprintf("failed to upsert MARTA API response to postgres: %s", err.Error()))
+			}
```

Comment:
If the record being upserted creates a new run, we're going to want to do directional correction. That requires being able to see the _other_ records in the batch that corresponded to the same train, so we group them all by train (and also by direction and line, which is unnecessary I guess but shouldn't hurt).